pkg/dynamicinstrumentation/ditypes/config.go

```diff
@@ -36,6 +36,12 @@ var (
 	SliceMaxLength          = 100   // SliceMaxLength is the default limit in number of elements of a slice
 )
 
+// RuntimeOptions contains configuration options for how the dynamic instrumentation product
+// should behave at runtime
+type RuntimeOptions struct {
+	ResolveInlinedProgramCounters bool
+}
+
```

Comment:
We are passing this through multiple layers and changing signatures of many functions to check it in a single place in analysis.

As far as I understand it, the problem now is the performance impact of extracting information about inline functions to put in the stacktrace is too high now given that we've also removed DWARF caching.

I don't see the point in having this configurable - either performance is good enough that we want it on for everyone or it's not in case we don't do it.

I would keep a local configuration in the analysis component that can be overridden for development and keep the rest of the code simple.

Reply:
Yea you make a good point, perhaps taking it a step further should I just remove it completely and we can readdress it in the future? When I originally developed this I wrote a [blog post](https://www.grant.pizza/blog/go-stack-traces-bpf/) about it, it's well documented enough that we can pick up where we left off even if it's not locally configurable at all.